aptos-move/framework/src/natives/aggregator_natives/helpers_v2.rs

```diff
@@ -13,14 +16,34 @@ use move_vm_types::values::{Struct, StructRef, Value};
 const VALUE_FIELD_INDEX: usize = 0;
 const LIMIT_FIELD_INDEX: usize = 1;
 
+pub fn u128_to_u64(value: u128) -> PartialVMResult<u64> {
+    u64::try_from(value).map_err(|_| {
+        PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+            .with_message("Cannot cast u128 into u64".to_string())
+    })
+}
+
 pub(crate) fn aggregator_value_field_as_id(value: u128) -> PartialVMResult<AggregatorID> {
-    if value > u64::MAX as u128 {
-        return Err(
-            PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
-                .with_message("Aggregator identifier is too small".to_string()),
-        );
+    u128_to_u64(value).map(AggregatorID::new)
+}
+
+pub(crate) fn aggregator_snapshot_value_field_as_id(
+    value: SnapshotValue,
+) -> PartialVMResult<AggregatorID> {
+    match value {
+        SnapshotValue::Integer(v) => aggregator_value_field_as_id(v),
+        SnapshotValue::String(v) => Ok(String::from_utf8(v)
+            .map_err(|e| {
+                PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+                    .with_message(format!("Unable to convert bytes to string: {}", e))
+            })?
+            .parse::<u64>()
```

Comment:
@georgemitenkov we talked about this :)

I wouldn't use from_le_bytes, because then the string we put there can be invalid UTF, and so if value ever gets validated(i.e. doing serialization) or printed - it will break.

if the performance matters, we can create our function to do so